File: server/ruby/app/infrastructure/mappers/payment_details/customer_bank_account_mapper.rb
module Infrastructure
  module Mappers
    module PaymentDetails
      class CustomerBankAccountMapper
        def self.from_sdk_response(response)
          return nil if response.nil?

          dto = Business::Domain::Payments::PaymentDetails::CustomerBankAccount.new
          dto.account_holder_name = response.account_holder_name
          dto.bic = response.bic
          dto.iban = response.iban

          dto
        end
      end
    end
  end
end

File: server/python/app/infrastructure/mappers/payment_details/customer_bank_account_mapper.py
from typing import Optional
from app.application.domain.payments.payment_details.customer_bank_account import CustomerBankAccount
from onlinepayments.sdk.domain.customer_bank_account import CustomerBankAccount as CustomerBankAccountSdk

class CustomerBankAccountMapper:

    @staticmethod
    def map_from_sdk_response(response: Optional[CustomerBankAccountSdk]) -> Optional[CustomerBankAccount]:
        if response is None:
            return None

        return CustomerBankAccount(
            account_holder_name = response.account_holder_name,
            bic = response.bic,
            iban = response.iban
        )